gno.land/cmd/gnoweb/main.go

```diff
@@ -18,6 +18,20 @@ import (
 	"go.uber.org/zap/zapcore"
 )
 
+// normalizeRemoteURL ensures the remote URL has a valid HTTP(S) protocol.
+// - tcp:// is converted to http:// (RPC uses HTTP over TCP)
+// - No protocol defaults to http://
+// - http:// and https:// are kept as-is
+func normalizeRemoteURL(remote string) string {
+	if strings.HasPrefix(remote, "tcp://") {
+		return strings.Replace(remote, "tcp://", "http://", 1)
+	}
+	if strings.HasPrefix(remote, "http://") || strings.HasPrefix(remote, "https://") {
+		return remote
+	}
+	return "http://" + remote
+}
+
```

Comment:
I think I would prefer something like this, explicitly panicking if the protocol is not supported. (i.e. unix in the browser should not be allowed)

```suggestion
	protocol, rest, found := strings.Cut(remote, "://")
	if !found {
		return "http://" + remote
	}
	
	switch protocol {
	case "tcp":
		return "http://" + rest
	case "http", "https":
		return remote
	default:
		panic("unsupported protocol: "+ protocol)
}
```